File: 卡码/代码随想录/Python/二叉树/06-二叉搜索树修改与构造/LeetCode_0669.py
#669. 修剪二叉搜索树
from typing import *
from ..TypeNode import *

class Solution:
  def trimBST(self, root: Optional[TreeNode], low: int, high: int) -> Optional[TreeNode]:
    if not root:
      return None
    if root.val<low:
      return self.trimBST(root.right,low,high)
    elif root.val>high:
      return self.trimBST(root.left,low,high)
    else:
      root.left=self.trimBST(root.left,low,high)
      root.right=self.trimBST(root.right,low,high)
    return root

File: 卡码/代码随想录/C++/二叉树/06-二叉搜索树修改与构造/LeetCode_0669.cpp
//669. ÐÞ¼ô¶þ²æËÑË÷Ê÷
#include <iostream>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
  TreeNode* trimBST(TreeNode* root, int low, int high) {
		if(!root) return nullptr;
		if(root->val<low) return trimBST(root->right,low,high);
		else if(root->val>high) return trimBST(root->left,low,high);
		else{
			root->left=trimBST(root->left,low,high);
			root->right=trimBST(root->right,low,high);
		}
		return root;
  }
};

int main(){
  system("pause");
  return 0;
}
